src/types/generator.ts

```diff
@@ -147,7 +147,11 @@ export type ClientHeaderBuilder = (params: {
   provideIn: boolean | 'root' | 'any';
 }) => string;
 
-export type ClientFooterBuilder = (operationIds?: string[]) => string;
+export type ClientFooterBuilder = (options: {
```

Comment:
it's a breaking change 🙁

Reply:
what we could do is instead of generating them directly in this file. We add them with the other models? WOuld be better and more consistent no?

Reply:
Indeed we could do that, but how would you get the operationName from there?

Reply:
this client footer builder is overridable here?

Reply:
yes

Reply:
You could generate the new models here but you can return an array of `GeneratorSchema` by the client. Like that they can be added to the other models/schemas

Reply:
Ok, sorry but I might need more explanations here. I don't understand why this is breaking, for me the footer builder is not exposed to the configuration itself, it seems to be used only internally. What did I missing?

Do you mean I should have a new prop `schemas` [https://github.com/CPatchane/orval/blob/de2e0e622cd24340bdbb4810180dd465256c9dfe/src/core/generators/client.ts#L258]?

Reply:
Look the type [here](https://github.com/anymaniax/orval/blob/master/src/types/index.ts#L42). You will see that the function type is exposed

Reply:
To avoid the breaking change you could keep the two `signature` and check when it's an array to handle it like before and add a warning no?

Reply:
It make things much more complicated to handle, using function overload will make types breaking since TS will expect to handle the two types of arguments in the custom footer function.
But I think this change is necessary since it make things more consistent between the footer and the header (both functions is using an options object as parameter) and more maintainable if we want to add more options in the future.
I am afraid I can't add something simple/not ugly to handle that case without that :/ (if you still want to do it I can push a commit that should work)

Reply:
@anymaniax Any updates on this 🙏 ?

Reply:
Sorry, I am a bit busy this week and next week. I will take a look as soon as possible

Reply:
I quick solution, for now, would be to have two arguments no? keep the operationIds in first and have the options in second. With a warning that it's deprecated for the next version?

Reply:
Indeed that could be a quick solution since the second argument would be optional then

Reply:
I'll rework using that quick fix then 👍

Reply:
It makes thing more difficult to maintain actually, you'll have to call the function each time wtih `undefined` as first argument.
What do you think of the last commit? If this is a custom client, I use a string arrays and if it works the developer will get a warn about that deprecation. Using that, it will be easier for you to remove that part when you'll release a new major version and you can use the new signature without any issues.